books/views.py

```diff
@@ -184,3 +186,11 @@ def user_page_widget(request, username):
     user = get_object_or_404(User, username=username)
     books = UserBook.objects.select_related('book').filter(user=user, status='c')
     return render(request, 'widget.html', {'books': books})
+
+@login_required
+def user_favorite(request):
+    user = request.user
+    book = request.GET.get('book')
+    checked = True if request.GET.get('checked') == "true" else False
+    UserBook.objects.filter(user__username=user,book__bookid=book).update(favorite=checked)
```

Comment:
shouldn't this be the user object? `user=user`?

Reply:
I would expect there always be one userbook? so should we use `get` instead of `filter`, that might express the intent better

Reply:
> shouldn't this be the user object? `user=user`?

Interesting!!! 
what i understand of Django ORM is that it adds a primary key on it's own(called id) if you don't specify explicitly.
Django ORM can combines table having foreign key constraint using join query or subquery but that will be based on these primary key by default. if we want to combine based on non primary field, we need to specify like table__field. 

But in case of User, it looks like it already have declared username as primary. that's why user=username works.
Let me know if i am wrong.

Reply:
> I would expect there always be one userbook? so should we use `get` instead of `filter`, that might express the intent better

i have fixed it. it is more useful in case where multiple different books with same bookid appears. this will raise exception and ajax query will not work. i guess  similar issue has been seen in issue #22